### PR TITLE
DISTX-197. Implement telemetry on env side

### PR DESCRIPTION
**Pull request by oleewere**
- added telemetry request / response on env side
- save global telemetry settings on env side (nullable - dynamic)
- merge global env telemetry settings with the stack level one

UI changes should be added in the future for the env API

No request validation yet (will be added in another PR) - overall every new field is nullsafe right now, should not broke anything
I will be out until next Thursday so no hurry with merging this

### CODE REVIEW

File: environment/src/main/java/com/sequenceiq/environment/credential/v1/converter/TelemetryRequestToTelemetryConverter.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import com.sequenceiq.environment.api.v1.environment.model.request.WorkloadAnalyticsRequest;
+
+@Component
+public class TelemetryRequestToTelemetryConverter {
+
+    public Telemetry convert(TelemetryRequest request) {
+        Telemetry telemetry = null;
+        if (request != null) {
+            Logging logging = null;
+            WorkloadAnalytics wa = null;
```
Comment (attilapalfi92) [inline environment/src/main/java/com/sequenceiq/environment/credential/v1/converter/TelemetryRequestToTelemetryConverter.java]: I know it's a local variable but it's still a too short name for it.

File: core/src/main/java/com/sequenceiq/cloudbreak/converter/v4/stacks/StackV4RequestToStackConverter.java [hunk 2]
```diff
@@ -199,30 +203,60 @@ private void updateCustomDomainOrKerberos(StackV4Request source, Stack stack) {
...
+            if (stackTelemetry.getLogging() != null) {
+                LOGGER.debug("Found stack level telemetry (logging) settings. It will override enviroment level telemetry configs.");
+                logging = stackTelemetry.getLogging();
+            }
+            if (stackTelemetry.getWorkloadAnalytics() != null) {
+                LOGGER.debug("Found stack level telemetry (workload analytics) settings. It will override enviroment level telemetry configs.");
+                wa = stackTelemetry.getWorkloadAnalytics();
+            }
+        }
+        return new Telemetry(logging, wa);
```
[the remark this answers is not in the document]
Comment (oleewere) [reply core/src/main/java/com/sequenceiq/cloudbreak/converter/v4/stacks/StackV4RequestToStackConverter.java]: yes that's ok. we can expect those values are null from the requests

File: core/src/main/java/com/sequenceiq/cloudbreak/converter/v4/stacks/StackV4RequestToStackConverter.java [hunk 3]
```diff
@@ -199,30 +203,60 @@ private void updateCustomDomainOrKerberos(StackV4Request source, Stack stack) {
...
             LOGGER.debug("Exception during reading telemetry settings.", e);
             throw new BadRequestException("Failed to convert dynamic telemetry settingss.");
         }
     }
 
+    @VisibleForTesting
+    Telemetry mergeGlobalAndStackLevelTelemetry(TelemetryResponse envTelemetry, Telemetry stackTelemetry) {
+        Logging logging = null;
+        WorkloadAnalytics wa = null;
+        if (envTelemetry != null) {
```
[the remark this answers is not in the document]
Comment (oleewere) [reply core/src/main/java/com/sequenceiq/cloudbreak/converter/v4/stacks/StackV4RequestToStackConverter.java]: i would not extend this to make the stack converter less readable (at least for my eyes), so I'm creating a util for the telemetry merger, will extend that one there

**Comment by oleewere** [comment]
retest this please

File: environment/src/main/java/com/sequenceiq/environment/credential/v1/converter/TelemetryRequestToTelemetryConverter.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+import com.sequenceiq.cloudbreak.cloud.model.Logging;
+import com.sequenceiq.cloudbreak.cloud.model.Telemetry;
+import com.sequenceiq.cloudbreak.cloud.model.WorkloadAnalytics;
+import com.sequenceiq.environment.api.v1.environment.model.request.LoggingRequest;
+import com.sequenceiq.environment.api.v1.environment.model.request.TelemetryRequest;
+import com.sequenceiq.environment.api.v1.environment.model.request.WorkloadAnalyticsRequest;
+
+@Component
+public class TelemetryRequestToTelemetryConverter {
```
Comment (attilapalfi92) [inline environment/src/main/java/com/sequenceiq/environment/credential/v1/converter/TelemetryRequestToTelemetryConverter.java]: Please merge this and `TelemetryToTelemetryResponseConverter` into a single class, called `TelemetryApiConverter`

File: environment/src/main/java/com/sequenceiq/environment/credential/v1/converter/TelemetryToTelemetryResponseConverter.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+import com.sequenceiq.cloudbreak.cloud.model.Logging;
+import com.sequenceiq.cloudbreak.cloud.model.Telemetry;
+import com.sequenceiq.cloudbreak.cloud.model.WorkloadAnalytics;
+import com.sequenceiq.cloudbreak.common.json.Json;
+import com.sequenceiq.environment.api.v1.environment.model.response.LoggingResponse;
+import com.sequenceiq.environment.api.v1.environment.model.response.TelemetryResponse;
+import com.sequenceiq.environment.api.v1.environment.model.response.WorkloadAnalyticsResponse;
+
+@Component
+public class TelemetryToTelemetryResponseConverter {
```
Comment (attilapalfi92) [inline environment/src/main/java/com/sequenceiq/environment/credential/v1/converter/TelemetryToTelemetryResponseConverter.java]: See previous comment

File: environment/src/main/java/com/sequenceiq/environment/environment/v1/EnvironmentApiConverter.java [hunk 6]
```diff
@@ -41,12 +43,20 @@
...
     public EnvironmentApiConverter(ThreadBasedUserCrnProvider threadBasedUserCrnProvider,
             RegionConverter regionConverter,
-            CredentialToCredentialV1ResponseConverter credentialConverter) {
+            CredentialToCredentialV1ResponseConverter credentialConverter,
+            TelemetryRequestToTelemetryConverter telemetryRequestConverter,
+            TelemetryToTelemetryResponseConverter telemetryResponseConverter) {
         this.threadBasedUserCrnProvider = threadBasedUserCrnProvider;
         this.regionConverter = regionConverter;
         this.credentialConverter = credentialConverter;
+        this.telemetryRequestConverter = telemetryRequestConverter;
```
Comment (attilapalfi92) [inline environment/src/main/java/com/sequenceiq/environment/environment/v1/EnvironmentApiConverter.java]: `TelemetryApiConverter` will be the only new dependency once you merged them.

**Pull request merged**